service/matching/tasklist/matcher.go

```diff
@@ -567,6 +567,63 @@ func (tm *TaskMatcher) pollNonBlocking(
 	taskC <-chan *InternalTask,
 	queryTaskC <-chan *InternalTask,
 ) (*InternalTask, error) {
+	waitTime := tm.config.LocalPollWaitTime()
+	if waitTime > 0 {
+		childCtx, cancel := context.WithTimeout(ctx, waitTime)
```

Comment:
nit: move it to a separate function since you never continue the code path after this if.

Also, it will allow us to test this behavior separately.